src/main/java/io/percy/selenium/Percy.java

```diff
@@ -164,6 +173,81 @@ public void snapshot(String name, Map<String, Object> options) {
+        Map<String, Object> options = new HashMap<String, Object>();
+        screenshot(name, options);
+    }
+
+    /**
+     * Take a snapshot and upload it to Percy.
+     *
+     * @param name      The human-readable name of the screenshot. Should be unique.
+     * @param options   Extra options
+     */
+    public void screenshot(String name, Map<String, Object> options) throws UnsupportedOperationException {
+        if (!isPercyEnabled) { return; }
+        List<String> driverArray = Arrays.asList(driver.getClass().toString().split("\\$")); // Added to handle testcase (mocked driver)
+        Iterator<String> driverIterator = driverArray.iterator();
+        String driverClass = driverIterator.next();
+        if (!driverClass.equals(RemoteWebDriver.class.toString())) { throw new UnsupportedOperationException(
+                String.format("Driver should be of type RemoteWebDriver, passed is %s", driverClass)
+        ); }
+
+        String sessionId = ((RemoteWebDriver) driver).getSessionId().toString();
+        CommandExecutor executor = ((RemoteWebDriver) driver).getCommandExecutor();
+
+        // Get HttpCommandExecutor From TracedCommandExecutor
+        if (executor instanceof TracedCommandExecutor) {
+            Class className = executor.getClass();
+            try {
+                Field field = className.getDeclaredField("delegate");
+                // make private field accessible
+                field.setAccessible(true);
+                executor = (HttpCommandExecutor)field.get(executor);
+            } catch (Exception e) {
+                log(e.toString());
+                return;
+            }
+        }
+        String remoteWebAddress = ((HttpCommandExecutor) executor).getAddressOfRemoteServer().toString();
+
+        Capabilities caps = ((RemoteWebDriver) driver).getCapabilities();
+        ConcurrentHashMap<String, String> capabilities = new ConcurrentHashMap<String, String>();
+        List<String> capsNeeded = new ArrayList<>(Arrays.asList("browserName", "platform", "version", "osVersion", "proxy"));
+        Iterator<String> iterator = capsNeeded.iterator();
+        while (iterator.hasNext()) {
+            String cap = iterator.next();
+            if (caps.getCapability(cap) != null) {
+                capabilities.put(cap, caps.getCapability(cap).toString());
+            }
+        }
+
+        // Build a JSON object to POST back to the agent node process
+        JSONObject json = new JSONObject(options);
+        json.put("sessionId", sessionId);
+        json.put("commandExecutorUrl", remoteWebAddress);
+        json.put("capabilities", capabilities);
+        json.put("snapshotName", name);
+        json.put("clientInfo", env.getClientInfo());
+        json.put("environmentInfo", env.getEnvironmentInfo());
+
+        StringEntity entity = new StringEntity(json.toString(), ContentType.APPLICATION_JSON);
+
+        try (CloseableHttpClient httpClient = HttpClientBuilder.create().build()) {
```

Comment:
update `postSnapshot` to allow below screenshot posting as well, don't duplicate posting to percy local server code

Reply:
Lets do this refactoring

Reply:
doing this refactoring means changing a lot of existing code.
is it ok if i create a new function instead for this

Reply:
done